base/src/Data/Macaw/AbsDomain/JumpBounds.hs

```diff
@@ -386,8 +386,18 @@ exprRangePred info e = do
             -- Compare the range constraint with the output number of bits.
             -- If the bound on r covers at most the truncated number
             -- of bits, then we just pass it through.
-          , Just LeqProof <- testLeq (rangeWidth r) w ->
-            SomeRangePred r
+          -> case testLeq (rangeWidth r) w of
+              Just LeqProof -> SomeRangePred r
+              
+              -- Otherwise, our bound remains valid for the truncated number
+              -- of bits and we can tighten the bounds to at most @2^w-1@ (inclusive)
```

Comment:
Worth mentioning in this comment that we may need to clamp the lower or upper bounds if the fall outside of the range `[0, 2^w-1]`.

Reply:
I think what I meant with "tighten the bounds" is too ambiguous.  How about the following:

> Otherwise, we need to rewrite the constraints to be on `w` bits. We can do that by clamping the lower/upper bounds if they're outside of the range `[0, 2^w-1]`.

Reply:
That sounds reasonable to me.

Reply:
Replaced with this comment.